crates/release_plz/src/args/mod.rs

```diff
@@ -84,6 +100,35 @@ fn parse_config(config_path: Option<&Path>) -> anyhow::Result<Config> {
     toml::from_str(&config).with_context(|| format!("invalid config file {config_path:?}"))
 }
 
+fn parse_config_from_metadata(project_manifest: &Path) -> anyhow::Result<Option<Config>> {
+    let metadata = cargo_metadata::MetadataCommand::new()
+        .no_deps()
+        .manifest_path(project_manifest)
+        .exec()
+        .context("failed to execute cargo_metadata")?;
```

Comment:
`workspace_members` is used for `update` subcommand and I'm not sure which way is the best for passing around the metadata object. Feel free to guide me on this!

Reply:
We can add an optional field to the `UpdateRequest`:
- If None, we run cargo metadata in the `update` subcommand (useful for external libraries using release_plz_core, that don't want to run cargo metadata by themselves.
- Otherwise, we use this field.

You can set this field here for example:
https://github.com/MarcoIeni/release-plz/blob/986bbc4c187258eb644f6b6361e3d25168975550/crates/release_plz/src/args/update.rs#L116

What do you think? 👀

Reply:
I tried to implement that and failed - it is still not clear to me. Also, we need to take into account that this functionality also effects `release` command so doing this only for `UpdateRequest` might not be enough.

Reply:
I started working on this in https://github.com/MarcoIeni/release-plz/pull/1167

`cargo check` works. I need to fix tests.

Reply:
Nice!

Reply:
https://github.com/MarcoIeni/release-plz/pull/1167 merged!
It was so painful to implement, you were right 😅 
But now you have the metadata field there!

Reply:
I realized `cargo_metadata` is available to `update_request` and `release_request` functions but we are parsing the configuration file in `config` function. How do you think I can re-use the metadata there?

Reply:
by passing the metadata as reference?
![image](https://github.com/MarcoIeni/release-plz/assets/11428655/369e654b-ea08-4239-9a36-455c46115c78)

Reply:
Thank you! Done in 3d5bf881d19969d89381a80a8a86fc34f244b205